master/internal/templates/api.go

```diff
@@ -285,3 +292,31 @@ func (a *TemplateAPIServer) DeleteTemplate(
 	}
 	return &apiv1.DeleteTemplateResponse{}, nil
 }
+
+func canCreateTemplateWorkspace(ctx context.Context, user *model.User, workspaceID int32) error {
+	err := workspace.AuthZProvider.Get().CanGetWorkspaceID(ctx, *user, workspaceID)
+	if err != nil {
+		return err
+	}
+
+	exists, err := workspace.Exists(ctx, int(workspaceID))
+	switch {
+	case err != nil:
+		return fmt.Errorf("failed to check workspace %d: %w", workspaceID, err)
```

Comment:
How about `workspace check failed` instead of `failed to check workspace`? I think it's a little simpler and more descriptive. When I first read `failed to check workspace` I thought it was saying the user had failed to do something and then I read it again.

Reply:
I see now that this function is not new as much as moved ... feel free to ignore if you'd prefer.